server/service/src/sync/translations/invoice_line.rs

```diff
@@ -81,57 +85,93 @@ impl SyncTranslation for InvoiceLineTranslation {
             return Ok(None);
         }
 
-        let data = serde_json::from_str::<LegacyTransLineRow>(&sync_record.data)?;
+        let LegacyTransLineRow {
+            id,
+            invoice_id,
+            item_id,
+            item_name,
+            stock_line_id,
+            location_id,
+            batch,
+            expiry_date,
+            pack_size,
+            cost_price_per_pack,
+            sell_price_per_pack,
+            r#type,
+            number_of_packs,
+            note,
+            item_code,
+            tax,
+            total_before_tax,
+            total_after_tax,
+        } = serde_json::from_str::<LegacyTransLineRow>(&sync_record.data)?;
 
-        let line_type = to_invoice_line_type(&data._type).ok_or(anyhow::Error::msg(format!(
+        let line_type = to_invoice_line_type(&r#type).ok_or(anyhow::Error::msg(format!(
             "Unsupported trans_line type: {:?}",
-            data._type
+            r#type
         )))?;
 
-        let (item_code, tax, total_before_tax, total_after_tax) = match data.item_code {
+        let (item_code, tax, total_before_tax, total_after_tax) = match item_code {
             Some(item_code) => {
                 // use new om_* fields
                 (
                     item_code,
-                    data.tax,
-                    data.total_before_tax.unwrap_or(0.0),
-                    data.total_after_tax.unwrap_or(0.0),
+                    tax,
+                    total_before_tax.unwrap_or(0.0),
+                    total_after_tax.unwrap_or(0.0),
                 )
             }
             None => {
-                let item = match ItemRowRepository::new(connection).find_one_by_id(&data.item_ID)? {
+                let item = match ItemRowRepository::new(connection).find_one_by_id(&item_id)? {
                     Some(item) => item,
                     None => {
                         return Err(anyhow::Error::msg(format!(
                             "Failed to get item: {}",
-                            data.item_ID
+                            item_id
                         )))
                     }
                 };
-                let total = total(&data);
+                let total_multiplier = match r#type {
+                    LegacyTransLineType::StockIn => cost_price_per_pack,
+                    LegacyTransLineType::StockOut => sell_price_per_pack,
+                    _ => 0.0,
+                };
+
+                let total = total_multiplier * number_of_packs as f64;
                 (item.code, None, total, total)
             }
         };
 
+        let (stock_line_id, location_id) = if is_active_record_on_site(
```

Comment:
This is the actual change

Reply:
It might not hurt to add your comment from the PR here?

`// When invoice lines are coming from another site, we don't get stock line and location, so foreign key constraint is violated, thus we want to set them to None if it's foreign site record. `

I'm assuming these records are being picked up by the `or_is_null` part of the changelog filter, because we can't set `name_id` and `store_id` for incoming invoice lines?

Reply:
🙏 ac6696d704e34b8745bb95d6be72ef432d7d112c

> I'm assuming these records are being picked up by the or_is_null part of the changelog filter, because we can't set name_id and store_id for incoming invoice lines ?

name_id and store_id are set for incoming invoice lines in changelog (since invoice would be inserted). They don't belong to this site though, and during push they are removed via: https://github.com/openmsupply/open-msupply/blob/ac6696d704e34b8745bb95d6be72ef432d7d112c/server/service/src/sync/remote_data_synchroniser.rs#L155-L158

Which needs a comment: b8185a2c83d94f281e253ccfa920175a0a4be5a0